staging/src/k8s.io/client-go/discovery/content_type.go

```diff
@@ -0,0 +1,124 @@
+)
+
+// sortedContentTypeParams encapsulated content type parameters, where
+// the discovery content-type parameters will be ordered when printed,
+// but where parameter ordering will not matter when comparing.
+type sortedContentTypeParams struct {
+	params map[string]string
+}
+
+// sortedKeys returns content-type parameters sorted so keys
+// "g" > "v" > "as". No deterministic ordering if keys are not
+// discovery content-type parameters {"g", "v", "as"}.
+func (p *sortedContentTypeParams) sortedKeys() []string {
+	keys := make([]string, len(p.params))
+	for key := range p.params {
+		keys = append(keys, key)
+	}
+	// Ordering: "g" > "v" > "as"
+	sort.Slice(keys, func(i, j int) bool {
+		if strings.EqualFold(keys[i], "g") {
+			return true
+		} else if strings.EqualFold(keys[i],"as") {
+			return false
+		} else if strings.EqualFold(keys[i], "v") && strings.EqualFold(keys[j], "as") {
+			return true
+		} else {
+			return false
+		}
+	})
+	return keys
+}
+
+// String prints the content-type params using the sortedKeys array to
+// print the params in the order of the "sortedKeys" field.
+func (p *sortedContentTypeParams) String() string {
+	keyValues := []string{}
+	for _, key := range p.sortedKeys() {
+		if value, ok := p.params[key]; ok {
+			keyValues = append(keyValues, key+"="+value)
+		}
+	}
+	return strings.Join(keyValues, ";")
+}
+
+// Equals returns true if the passed parameters map is the same as the
+// receivers parameters. NOTE: the ordering of the parametes does NOT matter.
+func (p *sortedContentTypeParams) Equals(params map[string]string) bool {
+	return reflect.DeepEqual(p.params, params)
+}
+
+// IsV1ContentType returns true if the passed content-type string is
+// discovery version 1 (e.g. "application/json"). NOTE: The content-type
+// parameters can be anything OTHER than v2beta1 parameters.
+// Examples:
+//
+//	"application/json" = true
+//	"application/json; charset=UTF-8" = true
+//	"application/json; g=apidiscovery.k8s.io;v=v2beta1;as=APIGroupDiscoveryList" = false
+func IsV1ContentType(contentType string) bool {
+	base, params, err := mime.ParseMediaType(contentType)
+	if err != nil {
+		return false
+	}
+	return AcceptV1 == base && !AcceptV2Beta1Params.Equals(params)
+}
+
+// IsV2Beta1ContentType checks of the content-type string is both
+// "application/json" and contains the v2beta1 content-type params.
+// NOTE: This function is resilient to the ordering of the
+// content-type parameters. Examples:
+//
+//	"application/json; g=apidiscovery.k8s.io;v=v2beta1;as=APIGroupDiscoveryList" = true
+//	"application/json; as=APIGroupDiscoveryList;v=v2beta1;g=apidiscovery.k8s.io" = true
+//	"application/json" = false
+//	"application/json; charset=UTF-8" = false
+func IsV2Beta1ContentType(contentType string) bool {
+	base, params, err := mime.ParseMediaType(contentType)
+	if err != nil {
+		return false
+	}
+	return AcceptV1 == base && AcceptV2Beta1Params.Equals(params)
```

Comment:
I think this should tolerate additional params like charset, narrowly checking the parameters we care about, more like this (with the inline strings made constants, probably):

```go
return AcceptV1 == base && 
  params["as"] == "APIGroupDiscoveryList" && 
  params["g"] == "apidiscovery.k8s.io" && 
  params["v"] == "v2beta1"
```